types.go

```diff
@@ -1012,14 +1012,14 @@ type TKEY struct {
 	Mode       uint16
 	Error      uint16
 	KeySize    uint16
-	Key        string
+	Key        string `dns:"size-hex:KeySize"`
 	OtherLen   uint16
-	OtherData  string
+	OtherData  string `dns:"size-hex:OtherLen"`
 }
 
 func (rr *TKEY) String() string {
 	// It has no presentation format
-	return ""
+	return fmt.Sprintf("%s Algo: %s KeyLen: %d OtherLen: %d", rr.Hdr.String(), rr.Algorithm, rr.KeySize, rr.OtherLen)
```

Comment:
If we going to output this, we should
a) make sure it is a standard
b) make sure we can parse it (from text)

Reply:
I added it for just being able to dump and debug it.  I can't find a well-known or standard format, if someone knows of one please point me to it.  I think that leaves us with three choices

1. Have no text representation (not a fan of that one because String() is useful for debug)
2. Come up with our own format and add the parser

Let me know what's preferred - I could try to add the parser.  Note that the current TSIG implementation is similar to this one.  There is a string representation but no parser.  The existing parser chokes on the TSIG string output when trying to parse it as 3597.